packages/client/hmi-client/src/components/model/petrinet/tera-distribution-input.vue

```diff
@@ -0,0 +1,197 @@
+						{{ hoveredOption?.name }}
+						<a
+							v-if="externaDoclLinks[hoveredOption.value]"
+							v-tooltip="hoveredOption.name + ' pytorch.org docs'"
+							:href="externaDoclLinks[hoveredOption.value]"
+							:aria-label="hoveredOption.name + ' docs'"
+							rel="noopener noreferrer"
+							target="_blank"
+						>
+							<span class="pi pi-external-link"></span>
+						</a>
+					</template>
+					<template #content>
+						{{ distributionDescription[hoveredOption?.value] }}
+						<h5 class="pt-3 pb-2">Required parameters:</h5>
+						<ul class="ml-5">
+							<li class="pb-2" v-for="(param, index) in DistributionInputLabels[hoveredOption.value]" :key="param">
+								<h6>{{ param }}</h6>
+								<p class="parameter-description">
+									{{ distributionParamDescription[DistributionInputOptions[hoveredOption.value][index]] }}
+								</p>
+							</li>
+						</ul>
+					</template>
+				</Card>
+			</section>
+		</section>
+		<template v-for="(option, index) in options" :key="index">
+			<tera-input-number
+				class="parameter-input"
+				error-empty
+				:label="DistributionInputLabels[parameter.type][index]"
+				:model-value="getParameterDistribution(modelConfiguration, parameterId, true)?.parameters[option]"
+				@update:model-value="onParameterChange($event, DistributionInputOptions[parameter.type][index])"
+			/>
+		</template>
+	</span>
+</template>
+<script setup lang="ts">
+import { computed, ref, onMounted, nextTick } from 'vue';
+import { getParameterDistribution, isNumberInputEmpty } from '@/services/model-configurations';
+import { Model, ModelConfiguration, ModelDistribution } from '@/types/Types';
+import Dropdown from 'primevue/dropdown';
+import TeraInputNumber from '@/components/widgets/tera-input-number.vue';
+import {
+	DistributionType,
+	externaDoclLinks,
+	distributionTypeOptions,
+	DistributionInputOptions,
+	DistributionInputLabels,
+	distributionDescription,
+	distributionParamDescription
+} from '@/services/distribution';
+import Card from 'primevue/card';
+
+const props = defineProps<{
+	model: Model;
+	modelConfiguration: ModelConfiguration;
+	parameterId: string;
+	parameter: ModelDistribution;
+}>();
+
+const emit = defineEmits(['update-parameter', 'update-source']);
+
+const options = computed(() => DistributionInputOptions[props.parameter.type]);
+
+const dropdownRef = ref();
+const cardRef = ref();
+const isParameterEmpty = ref<boolean>(false);
+const distributionOptions = ref<{ name: string; value: string }[]>(distributionTypeOptions());
+const hoveredOption = ref<{ name: string; value: string } | null>();
+const popupPosition = ref({ top: 0, left: 0 });
+
+function hidePopup() {
+	hoveredOption.value = null;
+}
+
+const cardElement = computed(() => cardRef.value.$el.getBoundingClientRect());
+
+async function showPopup(item: HTMLElement | null, index: number) {
+	hidePopup();
+	if (!item) return;
+
+	hoveredOption.value = distributionOptions.value[index];
+	await nextTick();
```

Comment:
Why do we need a `nextTick()` here?

Reply:
I'm waiting for the dropdown to finish rendering before using the `getBoundingClientRect()` to get sizing and position.

Reply:
Okay, we should add a comment when using "nextTick()" because it is a bit unusual and at time difficult for the readers to know what is the significance of the call.

Reply:
I get a bunch of warnings when I click the dropdown to change distribution types (eg from constant to uniform).

```
chunk-JRQYYDJE.js?v=1f88d7d8:2113 [Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement. 
  at <VirtualScroller ref=fn<bound virtualScrollerRef> items= 
(2) [{…}, {…}]
 style= 
{height: '200px'}
  ... > 
  at <BaseTransition onEnter=fn onAfterEnter=fn<bound onOverlayAfterEnter> onLeave=fn<onLeave>  ... > 
  at <Transition name="p-connected-overlay" onEnter=fn<bound onOverlayEnter> onAfterEnter=fn<bound onOverlayAfterEnter>  ... > 
  at <Portal appendTo="body" > 
  at <Dropdown id="beta" ref="dropdownRef" model-val
```

And changing the dropdown type seem to be ineffective, that is to say changing the distribution type doesn't seem to do anything. Are you seeing this as well?

Reply:
I'm not seeing this error

Reply:
I am still seem them, albeit they are now different. I am also seeing a bunch of errors.
Try opening this when you are doing `yarn stating`

And after I open the model-configuration modal I can no longer close it...

```
http://localhost:8080/projects/0f476a43-f7d4-433b-bee6-e853dba06522/workflow/a6efdfad-1ee4-43d5-ac18-bd312002e3e4?operator=1e618b6a-d0bc-48be-bb66-198c69797c3e
```

Errors:
```
chunk-JRQYYDJE.js?v=4f7b7627:6990 Uncaught (in promise) TypeError: Cannot set properties of null (setting '__vnode')
    at patchElement (chunk-JRQYYDJE.js?v=4f7b7627:6990:18)
    at processElement (chunk-JRQYYDJE.js?v=4f7b7627:6873:7)
    at patch (chunk-JRQYYDJE.js?v=4f7b7627:6730:11)
    at patchBlockChildren (chunk-JRQYYDJE.js?v=4f7b7627:7096:7)
    at patchElement (chunk-JRQYYDJE.js?v=4f7b7627:7014:7)
    at processElement (chunk-JRQYYDJE.js?v=4f7b7627:6873:7)
    at patch (chunk-JRQYYDJE.js?v=4f7b7627:6730:11)
    at ReactiveEffect.componentUpdateFn [as fn] (chunk-JRQYYDJE.js?v=4f7b7627:7450:9)
    at ReactiveEffect.run (chunk-JRQYYDJE.js?v=4f7b7627:479:19)
    at ReactiveEffect.runIfDirty (chunk-JRQYYDJE.js?v=4f7b7627:517:12)
```

Warnings:
```
[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement. 
  at <TeraModelPart class="pl-4" items= [{…}] feature-config= {isPreview: true} > 
  at <AccordionTab header="Observables" > 
  at <BaseTransition appear=false persisted=false mode=undefined  ... > 
  at <Transition name="p-toggleable-content" data-pc-section="accordiontab.transition" > 
  at <Accordion key=2 active-index= [0] > 
  at <TeraDrilldownSection is-loading=false tabName="Wizard" class="px-3 mb-10"  ... > 
  at <TeraColumnarPanel class="flex-1" > 
  at <TeraDrilldown downstream-operators-nav= [{…}] spawn-animation="scale" upstream-operators-nav= [{…}]  ... > 
  at <TeraModelConfigDrilldown key=0 downstream-operators-nav= [{…}] node= {id: '1e618b6a-d0bc-48be-bb66-198c69797c3e', workflowId: 'a6efdfad-1ee4-43d5-ac18-bd312002e3e4', version: 15, isDeleted: false, createdBy: 'adam', …}  ... > 
  at <TeraWorkflow key=5 asset-id="a6efdfad-1ee4-43d5-ac18-bd312002e3e4" > 
  at <TeraProject projectId="0f476a43-f7d4-433b-bee6-e853dba06522" pageType="workflow" assetId="a6efdfad-1ee4-43d5-ac18-bd312002e3e4"  ... > 
  at <RouterView> 
  at <App>
```